app/src/kscan.c

```diff
@@ -47,14 +47,19 @@ void zmk_kscan_process_msgq(struct k_work *item) {
 
     while (k_msgq_get(&zmk_kscan_msgq, &ev, K_NO_WAIT) == 0) {
         bool pressed = (ev.state == ZMK_KSCAN_EVENT_STATE_PRESSED);
-        uint32_t position = zmk_matrix_transform_row_column_to_position(ev.row, ev.column);
-        LOG_DBG("Row: %d, col: %d, position: %d, pressed: %s", ev.row, ev.column, position,
-                (pressed ? "true" : "false"));
-        ZMK_EVENT_RAISE(new_zmk_position_state_changed(
-            (struct zmk_position_state_changed){.source = ZMK_POSITION_STATE_CHANGE_SOURCE_LOCAL,
-                                                .state = pressed,
-                                                .position = position,
-                                                .timestamp = k_uptime_get()}));
+        int32_t position = zmk_matrix_transform_row_column_to_position(ev.row, ev.column);
+        if (position < 0) {
+            LOG_WRN("Not found in transform: row: %d, col: %d, pressed: %s", ev.row, ev.column,
+                    (pressed ? "true" : "false"));
```

Comment:
Is the pressed/released info relevant here?

Reply:
This matches the LOG_DBG for when something is found in the transform (except for the position, which cannot be printed if not found). I think if we assume that this can be triggered only by an incorrect transform, then the user won't care if it's a press/release event. If this is however triggered by a kscan driver that spits out wrong data,  or if it's triggered by a damaged matrix, then the user may find the pressed/released information useful. Imagine someone debugging a short in their keyboard matrix, and they are getting a release event when they put pressure on the area of the keyboard, knowing it was a release could be useful. I'm advocating for keeping it, but if you want I can remove it.

Reply:
I agree on keeping it.